clients/vault/src/system.rs

```diff
@@ -329,11 +329,22 @@ impl VaultService {
 		shutdown: ShutdownSender,
 	) -> Result<Self, Error> {
 		let is_public_network = spacewalk_parachain.is_public_network().await?;
+		let cfg_string = fs::read_to_string(&config.stellar_overlay_config_filepath)?;
+		let cfg: StellarOverlayConfig = serde_json::from_str(&cfg_string)
```

Comment:
Would it maybe make sense to store this created `StellarOverlayConfig` in the `VaultService` struct, similar to how we store the `VaultServiceConfig` in it? Just thought about it because in the code changes you added to another function below we go through creating it from the path again, although we could also just re-use the config we create here.